vars/centralPipelineLoad.groovy

```diff
@@ -0,0 +1,45 @@
+import com.sap.piper.Utils
+
+/**
+ * centralPipelineLoad
+ * Load a central pipeline.
+ *
+ */
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: 'centralPipelineLoad', stepParameters: parameters) {
+
+        def utils = new Utils()
+
+        // The coordinates of the central pipeline script
+        def repo = utils.getMandatoryParameter(parameters, 'repoUrl', null)
+        def branch = utils.getMandatoryParameter(parameters, 'branch', 'master')
+        def path = utils.getMandatoryParameter(parameters, 'jenkinsfilePath', 'Jenkinsfile')
+
+        // In case access to the repository containing the central pipeline
+        // script is restricted the credentialsId of the credentials used for
+        // accessing the repository needs to be provided below. The corresponding
+        // credentials needs to be configured in Jenkins accordingly.
+        def credentialsId = utils.getMandatoryParameter(parameters, 'credentialsId', '')
+
+        // temporary folder used for storing the central jenkins file locally.
+        def temporaryPipelineFolder = "pipeline-${UUID.randomUUID().toString()}"
```

Comment:
any specific reason that the file is put into a temporary folder? I think this could even be omitted. You anyway work on an empty directory where the file is checked out to.

Reply:
IMO we should avoid clashes. E.g.: we sync Jenkinsfile from remote, having it as Jenkins file in the workspace root. Afterward we checkout the project sources, and maybe we have another Jenkinsfile there. With a tmp folder we are always on the save side.
We cannot assume that the payload repo is not syced into workspace root, e.g. into a ```scr```directory.

Other options would be to materialize the Jenkinsfile into ```java.io.tmpdir```.

Reply:
Exactly, this would be my recommendation. The pipeline would run on other nodes and therefore no clash. 
Anyway, after the file has been loaded there should not be any access to it anymore, IMHO.

Reply:
>we could also deleteDir() again right after loading ...

Disadvantage: in case of troubleshooting we do not have the file anymore. I mean, basically we know from the coordinates which version we have, but IMO we are in a better state if we have the file directly.

It does not harm if we keep the file.

Reply:
materializing the pipeline from remote is now done in a dedicated node() { ./. }